libs/execution_context/ebpf_core.c

```diff
@@ -865,6 +865,67 @@ _ebpf_core_protocol_map_get_next_key(
     EBPF_RETURN_RESULT(retval);
 }
 
+static ebpf_result_t
+_ebpf_core_protocol_program_test_run(
+    _In_ const ebpf_operation_program_test_run_request_t* request,
+    _Inout_updates_bytes_(reply_length) ebpf_operation_program_test_run_reply_t* reply,
+    uint16_t reply_length)
+{
+    EBPF_LOG_ENTRY();
+
+    ebpf_program_test_run_options_t options = {0};
+
+    ebpf_result_t retval;
+    ebpf_program_t* program = NULL;
+    size_t data_in_end;
+
+    // Validate that the request is large enough to contain the context_offset.
+    retval = ebpf_safe_size_t_add(
+        EBPF_OFFSET_OF(ebpf_operation_program_test_run_request_t, data), request->context_offset, &data_in_end);
+
+    if (retval != EBPF_SUCCESS)
+        goto Done;
+
+    if (data_in_end > request->header.length) {
+        retval = EBPF_INVALID_ARGUMENT;
+        goto Done;
+    }
+
+    retval =
+        ebpf_object_reference_by_handle(request->program_handle, EBPF_OBJECT_PROGRAM, (ebpf_core_object_t**)&program);
+    if (retval != EBPF_SUCCESS)
+        goto Done;
+
+    options.data_size_in = request->context_offset;
+    options.context_size_in = (size_t)(request->header.length) - data_in_end;
+    options.context_size_out = options.context_size_in;
```

Comment:
Will `context_size_out` always be same as `context_size_in`? Is it guaranteed both will be same size?